### PR TITLE
Update cypress to use centos docker image

### PR DESCRIPTION
**Pull request by cwdavies**
The old cypress docker image used Debian as base image, but the bureau policy only allows centos or alpine
The new cypress docker image is now based on centos with nodejs and chrome browser
Cypress has been updated to 7.2.0 in `package.json` and `yarn.lock`

## Additions

- Created a cypress `Dockerfile` based on centos that installs nodejs and Google chrome browser
- Added a cypress user and removed files in `Dockerfile` that cause vulnerability scans to fail

## Changes

- Updated Jenkinsfile to use cypress centos image pushed to docker registry
- Modified docker-compose.e2e.yml to use cypress centos docker image
- Updated Jenkinsfile to only build cypress image if it is not already in docker registry
- Added `try / finally` statements to free docker resources used
- Updated Jenkinsfile to run cypress docker containers as non-root user

## How to test this PR

1. docker-compose -f docker-compose.e2e.yml run admin-tests
2. docker-compose -f docker-compose.e2e.yml run component-tests
3. docker-compose -f docker-compose.e2e.yml run consumer-tools-tests
4. docker-compose -f docker-compose.e2e.yml run data-research-tests
5. docker-compose -f docker-compose.e2e.yml run paying-for-college-tests
6. docker-compose -f docker-compose.e2e.yml run rules-policy-tests

OR

`docker-compose -f docker-compose.e2e.yml up`

## Checklist
- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -95,10 +95,12 @@ pipeline {
...
-                    docker.build(env.IMAGE_NAME_LOCAL, '--build-arg scl_python_version=rh-python36 --target cfgov-prod .')
-                    docker.build(env.IMAGE_NAME_CYPRESS_LOCAL, '-f ./docker/cypress/Dockerfile .')
-                    docker.build(env.IMAGE_NAME_ES2_LOCAL, '-f ./docker/elasticsearch/Dockerfile .')
-                    docker.build(env.IMAGE_NAME_ES_LOCAL, '-f ./docker/elasticsearch/7/Dockerfile .')
+                    docker.withRegistry(dockerRegistry.url, dockerRegistry.credentialsId) {
+                        docker.build(env.IMAGE_NAME_CYPRESS_LOCAL, '--no-cache --pull -f ./docker/cypress/Dockerfile .') 
+                        docker.build(env.IMAGE_NAME_LOCAL, '--build-arg scl_python_version=rh-python36 --target cfgov-prod .')
+                        docker.build(env.IMAGE_NAME_ES2_LOCAL, '--no-cache --pull -f ./docker/elasticsearch/Dockerfile .')
+                        docker.build(env.IMAGE_NAME_ES_LOCAL, '--no-cache --pull -f ./docker/elasticsearch/7/Dockerfile .')
+                    }
```
Comment (Scotchester) [inline Jenkinsfile:103]: I am already doing this in #6396.

Comment (cwdavies) [reply Jenkinsfile]: Thanks for letting me know.

**Review by higs4281 — APPROVED** [review]
Hope I'm not stepping on final debugging, but these containers built and ran successful tests singly or in parallel, which gave my laptop an aerobic workout.

**Comment by Scotchester** [comment]
Thanks, @higs4281, but we have a bit of a roadblock with getting an initial upload of the ES7 or Cypress images up to DTR. See convo in ~cfgov-deployments.

**Comment by cwdavies** [comment]
> Thanks, @higs4281, but we have a bit of a roadblock with getting an initial upload of the ES7 or Cypress images up to DTR. See convo in ~cfgov-deployments.

@higs4281, I have a solution to get an initial upload of the ES7 and Cypress images into DTR.
I modified the Jenkinsfile to check if the docker tags already exist in DTR.
ES7 and Cypress docker images have different tags.
The docker images should only be pushed to DTR if the ES7 or Cypress images tags do not exist in DTR.

**Comment by cwdavies** [comment]
> Thanks, @higs4281, but we have a bit of a roadblock with getting an initial upload of the ES7 or Cypress images up to DTR. See convo in ~cfgov-deployments.

@higs4281, I have a solution to get an initial upload of the ES7 and Cypress images into DTR.
I modified the Jenkinsfile to check if the docker tags already exist in DTR.
ES7 and Cypress docker images have different tags.
The docker images should only be pushed to DTR if the ES7 or Cypress images tags do not exist in DTR.

**Pull request merged**